awx/main/models/events.py

```diff
@@ -586,6 +604,23 @@ def _update_host_summary_from_stats(self, hostnames):
     def job_verbosity(self):
         return self.job.verbosity
 
+    # If the Django primary key and the partition key have values, try update_or_create()
+    # Otherwise, fallback to inherited save()
+    def save(self, *args, **kwargs):
+        if getattr(self, self.PartitionInfo.partition_key, None) is None:
+            setattr(self, self.PartitionInfo.partition_key, datetime.datetime.now())
+
+        if getattr(self, "created", None) is None:
+            setattr(self, "created", datetime.datetime.now())
+            setattr(self, "modified", self.created)
+        else:
+            setattr(self, "modified", datetime.datetime.now())
```

Comment:
we already do a version of this in another module

https://github.com/ansible/awx/blob/devel/awx/main/dispatch/worker/callback.py#L163

so I want to first understand what about that has not worked for you. Did you find events without these fields unset? Could you get the data for those?

Reply:
The JobEvent model has been restored.